### PR TITLE
RANGE=THISANDFUTURE implementation

### PR DESCRIPTION
**Pull request by lightsofapollo**
Still rough and needs more tests / timezone sanity verification. I think the optimizations in duration may have not been completely correct as they _do_ account for timezones where I think the intent was to ignore those differences... I will look into it more and add more tests and then this should be good.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
@kewisch Do you want to review? I cleaned up the test/ directory a bit as well here.

**Comment by lightsofapollo** [comment]
@KevinGrandon You have some bandwidth to review? This will end up as a gaia patch but will require no changes aside from update ical.js so all the good stuff is here.

File: lib/ical/event.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
 ICAL.Event = (function() {
 
+  var THIS_AND_FUTURE = 'THISANDFUTURE';
```
Comment (KevinGrandon) [inline lib/ical/event.js]: Never appears to be used, remove? (nit)

**Comment by KevinGrandon** [comment]
Didn't test it, but I ran through the code and it makes sense to me. I found a single nit above, but it looks good. Nice tests too as always.

r=me

**Comment by lightsofapollo** [comment]
@KevinGrandon Thanks! fixed nits landing...

**Pull request merged**